### PR TITLE
MA-2000: Add custom related names to models.

### PR DESCRIPTION
**Pull request by jcdyer**
Please review @clintonb @nasthagiri 

This is to merge back to the edx branch (which I think is our main branch for this repo), and then to deploy to pypi.  @clintonb Please help me out with the pypi deployment process.

### CODE REVIEW

**Comment by clintonb** [comment]
@jcdyer once you've updated setup.py and merged, create a new tag to initiate a build/PyPI deployment.

**Comment by jcdyer** [comment]
@clintonb setup.py pulls the version number from `__init__.py`, so I think that's the only update it needs, yes?

**Comment by clintonb** [comment]
Yep, you're all set.

File: provider/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '0.5.0'
+__version__ = '0.6.0'
```
Comment (nasthagiri) [inline provider/__init__.py]: Since this is a backward incompatible change, should we update the major version to 1?

**Comment by nasthagiri** [comment]
Are there any repos other than edX-platform that depend on this library?

File: provider/__init__.py (hunk 1 above)
Comment (jcdyer) [inline provider/__init__.py]: I've always understood bumping the minor version to be acceptable for breaking changes in pre-1.0 software, but I don't have a strong feeling about the version number.

Comment (jcdyer) [inline provider/__init__.py]: See for instance, [Semantic Versioning Spec, Item 4](http://semver.org/#spec-item-4).

**Comment by jcdyer** [comment]
I'm fairly certain there are no other (internal) repos that depend on our version of django-outh2-provider.  edx-platform is our only oauth provider at the moment.

**Comment by nasthagiri** [comment]
:+1:

File: provider/__init__.py (hunk 1 above)
Comment (clintonb) [inline provider/__init__.py]: This is the first I have heard of Item 4. Considering we have no clear roadmap for when we will reach 1.x (for this, and pretty much all of our other libraries), I would much rather err on the side of caution and make this a major version bump.

Comment (jcdyer) [inline provider/__init__.py]: OK.  I'll make it version 1.0.0, then.

**Comment by jcdyer** [comment]
@clintonb Any other review comments?

**Comment by clintonb** [comment]
:+1:

**Pull request merged**